### PR TITLE
 Add nameSuffix field to kustomization.yaml

### PR DESCRIPTION
**Pull request by zoncoen**
ref. https://github.com/kubernetes-sigs/kustomize/pull/548

### CODE REVIEW

**Comment by zoncoen** [comment]
Add some documents in next PR!

File: pkg/commands/edit/set/set_name_suffix.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+func newCmdSetNameSuffix(fsys fs.FileSystem) *cobra.Command {
+	var o setNameSuffixOptions
+
+	cmd := &cobra.Command{
+		Use:   "namesuffix",
+		Short: "Sets the value of the nameSuffix field in the kustomization file.",
+		Example: `
+The command
+  set namesuffix -acme
+will add the field "nameSuffix: -acme" to the kustomization file if it doesn't exist,
```
Comment (Liujingfang1) [inline pkg/commands/edit/set/set_name_suffix.go]: I'm curious when you run `kustomize edit set namesuffix  -acme`, will it be treated as a flag?

Comment (zoncoen) [reply pkg/commands/edit/set/set_name_suffix.go]: Wow, that is correct. Fixed the example!

**Pull request merged**